lib/services/reconciler.go

```diff
@@ -197,7 +202,15 @@ func (r *Reconciler[T]) processNewResource(ctx context.Context, currentResources
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	if CompareResources(newT, registered) != Equal {
+
+	var resourcesEqual bool
+	hasEqual, ok := any(newT).(IsEqual)
+	if ok {
+		resourcesEqual = hasEqual.IsEqual(registered)
```

Comment:
Why not do this check inside `CompareResources`? Then everything that uses this method would benefit from speedier comparison as long as resources implement provided interface without having to check each call place.